sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -156,37 +164,41 @@ private HttpURLConnection connect(HttpRequest httpRequest) {
      * @param connection The HttpURLConnection that is being sent to
      */
     private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLConnection connection) {
-        BinaryData binaryDataBody = httpRequest.getBody();
+        BinaryData body = httpRequest.getBody();
 
-        if (binaryDataBody != null) {
-            switch (httpRequest.getHttpMethod()) {
-                case GET:
-                case HEAD: {
-                    return;
-                }
-                case OPTIONS:
-                case TRACE:
-                case CONNECT:
-                case POST:
-                case PUT:
-                case DELETE: {
-                    connection.setDoOutput(true);
-
-                    byte[] bytes = binaryDataBody.toBytes();
-
-                    try (DataOutputStream os = new DataOutputStream(
-                        new BufferedOutputStream(connection.getOutputStream()))) {
-                        os.write(bytes);
-                        os.flush();
-                    } catch (IOException e) {
-                        throw LOGGER.logThrowableAsError(new RuntimeException(e));
-                    }
+        if (body != null) {
+            HttpMethod method = httpRequest.getHttpMethod();
+            if (METHODS_WITHOUT_BODY.contains(method)) {
```

Comment:
nit: I think it's more readable, but probably nothing can bit perf of a good old switch and maybe we can get both of both worlds with `boolean canHaveBody(HttpMethod)`  helper?

Reply:
Yeah, this was me being a bit lazy here to get this more readable and easier to work with. Will change this back to either a `switch` or `if` check.